src/charm.py

```diff
@@ -30,105 +21,63 @@ class CephFSServerProxyCharm(ops.CharmBase):
-            logger.warning("Deferring ShareRequested event because filesystem info is not set")
-            self.unit.status = ops.BlockedStatus("No configured filesystem info")
-            event.defer()
-            return
-        share_info, auth_info = mount_info
-        logger.debug(f"CephFS mount info set to {share_info}")
-        self._cephfs_share.set_share(event.relation.id, share_info=share_info, auth_info=auth_info)
-
-    @property
-    def mount_info(self) -> Optional[Tuple[CephFSShareInfo, CephFSAuthInfo]]:
-        """Get the required info to mount the proxied CephFS share."""
-        if not (config := self.get_state("config")):
+        # Expected format: <filesystem_name>:<filesystem_shared_path>
+        sharepoint_cfg = str(config["sharepoint"])
+        sharepoint = sharepoint_cfg.split(":", maxsplit=1)
+        if len(sharepoint) != 2 or not sharepoint[0] or not sharepoint[1]:
+            msg = f"Invalid sharepoint `{sharepoint_cfg}`"
+            logger.error(msg)
+            self.unit.status = ops.BlockedStatus(msg)
             return
+        name, path = sharepoint
 
-        share_info = CephFSShareInfo(
-            fsid=config["fsid"],
-            name=config["name"],
-            path=config["path"],
-            monitor_hosts=config["monitor_hosts"],
-        )
-        auth_info = CephFSAuthInfo(username=config["username"], key=config["key"])
-
-        return (share_info, auth_info)
-
-    @property
-    def peers(self) -> Optional[ops.Relation]:
-        """Fetch the peer relation."""
-        return self.model.get_relation(PEER_NAME)
-
-    def get_state(self, key: str) -> dict[Any, Any]:
-        """Get a value from the global state."""
-        if not self.peers:
-            return {}
+        # Expected format: <ip/hostname>:<port> <ip/hostname>:<port>
+        monitor_hosts = str(config["monitor-hosts"]).split()
 
-        data = self.peers.data[self.app].get(key, "{}")
-        return json.loads(data)
+        # Expected format: <username>:<cephx-base64-key>
+        auth_info_cfg = str(config["auth-info"])
+        auth_info = auth_info_cfg.split(":", maxsplit=1)
+        if len(auth_info) != 2 or not auth_info[0] or not auth_info[1]:
+            msg = f"Invalid auth-info `{auth_info_cfg}`"
+            logger.error(msg)
+            self.unit.status = ops.BlockedStatus(msg)
+            return
+        user, key = auth_info
 
-    def set_state(self, key: str, data: Any) -> None:
-        """Insert a value into the global state."""
-        if not self.peers:
-            raise CharmError(
-                "Peer relation can only be accessed after the relation is established"
+        self._filesystem.set_info(
+            CephfsInfo(
+                fsid=fsid, name=name, path=path, monitor_hosts=monitor_hosts, user=user, key=key
             )
+        )
 
-        self.peers.data[self.app][key] = json.dumps(data)
+        self.unit.status = ops.ActiveStatus("Exporting share")
```

Comment:
Same as my comment on your nfs-server-proxy-operator PR. We should have the active status message be blank if everything is all good - e.g. the CephFS share is being exported - to reduce visual clutter in the `juju status` output. That way it's easier for admins to find problems, especially if there's a lot of charms in the deployment:

```suggestion
        self.unit.status = ops.ActiveStatus()
```